README.md

```diff
@@ -25,115 +28,261 @@ around**. Let’s take a deeper look at the bolded words:
+- The `main` method declares the variable `BinaryFunction sumSquares` and assigns the lambda expression `(x, y) -> x*x + y*y` to the variable.
+- The lambda expression implements the `apply` method, since that is the only abstract method in the `BinaryFunction` interface.
+  - The lambda expression's parameter types are inferred from the `apply` method.
+  - The lambda expression's return type is inferred from the `apply` method.
+- The method call `sumSquares.apply(2,3)` evaluates the expression as `2*2 + 3*3` and returns the result `13`.
 
-The lambda expression is associated with the body of the abstract function in
-the functional interface. We don’t need to specify any return type for the
-lambda expression since it can infer the type from the abstract method (we’ll
-discuss this in detail later). It can do this because functional interfaces must
-have only one abstract method and since a lambda expression is defining the body
-of that method there’s no ambiguity.
+An example of a lambda expression requiring a code block is shown below.
+A conditional statement is not an expression, so it must be enclosed within curly braces to create a code block.
+The `return` statements are required as the `apply` method signature indicates a return type `int`.
 
-If we need to add multiple lines to the lambda expression definition, we can use
-`{}` to wrap them.
+```java
+public class Example2 {
+    public static void main(String[] args) {
+        BinaryFunction min = (x, y) ->
+            {
+                if (x < y)
+                    return x ;
+                else
+                    return y;
+            };
+        System.out.println( min.apply(5, 12) );  //5
+        System.out.println( min.apply(23, 14) );   //14
+    }
+}
+```
+
+The conditional statement can be rewritten using the ternary operator `?`.
+Thus, we can rewrite the lambda expression to avoid the code block:
+
+```java
+BinaryFunction min = (x, y) -> (x < y) ? x : y;
+```
+
+## Redundant Code
+
+Consider the functional interface `StringOperator`, which transforms the string passed as a parameter:
 
 ```java
 @FunctionalInterface
-interface Greeter {
-    void display();
+public interface StringOperator {
+    String apply(String s);
 }
+```
+
+A lambda expression is a concise way of expressing a function in Java.  What does this mean?
+Let's consider two classes that implement `StringOperator` without lambda expressions. 
 
-public class Example {
+- `QuestionOperator` appends "?" to the string.
+- `CapitalizeOperator` converts the first letter in the string to uppercase.
+
+```java
+public class QuestionOperator implements StringOperator {
+    @Override
+    public String apply(String s) {
+        return s + "?";
+    }
+}
+
+public class CapitalizeOperator implements StringOperator {
+    @Override
+    public String apply(String s) {
+        return s.isEmpty()? s : s.substring(0,1).toUpperCase() + s.substring(1);
```

Comment:
Smallest little nit: add space between `s.isEmpty()` and `?` to conform with other format above.

Reply:
thanks, I updated the readme to add space.